basics/a40.go

```diff
@@ -0,0 +1,31 @@
+// defining a bunch of a different type of variables
+
+package main
+
+import "fmt"
+
+func main() {
+	var (
+		i int
+		f float64
+		s string
+	)
+
+	var (
+		processor_family = "Comet_lake"
+		cores = 8
+		base_frequency = 2.4
+	)
+	fmt.Println(i)
+	fmt.Println(f)
+	fmt.Println(s)
+	fmt.Println(processor_family)
+	fmt.Printf("Type of varilable processor_family = %T\n", processor_family)
+	fmt.Println(cores)
+	fmt.Printf("Type of varilable cores = %T\n", cores)
+	fmt.Println(base_frequency)
+	fmt.Printf("Type of base_frequency = %T\n", base_frequency)
+}
+
```

Comment:
I see the following output:

`
Comet lakey
Type of varilable family = string
8
Type of varilable cores = int
16
Type of varilable threads = int
2.4
Type of varilable frequency = float64
`

Could you add some comments to this program explaining a bit.

Reply:
ok